### PR TITLE
Exception_Handling_And_Structured

### PR DESCRIPTION
**Pull request by alidurrani1**
Exception Handling on API.
Minor changes related pep8.

### CODE REVIEW

File: app/api.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import json
 import requests
 import urllib
+import logging
```
Comment (maimoonaiqbal94) [inline app/api.py]: imports should be sorted alphabetically.

File: app/api.py [hunk 2]
```diff
@@ -17,11 +18,19 @@ def fetch_from_api():
...
+        'X-Parse-REST-API-Key': '9gA1WmxS3RolZChj4tgHkHj3Firlxt6Yn16DsTEi'
     }
-    data = json.loads(
-        requests.get(url, headers=headers).content.decode('utf-8'))  # Here you have the data that you need
+
+    try:
+        data = json.loads(
+            requests.get(url, headers=headers).content.decode('utf-8'))  # Here you have the data that you need
+    except requests.exceptions.HTTPError as err:
+        logging.critical('Bad Request Code')
```
Comment (maimoonaiqbal94) [inline app/api.py]: try to look into how HTTPErrors and their response codes are handled efficiently. and if the request fails due to some reason, also mention the reason.

File: app/views.py [hunk 3]
```diff
@@ -116,34 +125,38 @@ def home():
...
+                    session_token, "secret", algorithms=['HS256'])
+                logging.info('User succesfully logged in and is on Session Token')
 
                 # For email Sending  Just Uncomment And Put Password in email_sending() Function
                 # email_sending(link = 'http://127.0.0.1:5000/api?token='+session_token)
 
                 return redirect('api?token=' + session_token)
             except:
-
+                logging.info('User is shifted to Refresh Token')
```
Comment (maimoonaiqbal94) [inline app/views.py:137]: add meaningful logs, which user?

File: app/api.py (hunk 1 above)
Comment (alidurrani1) [reply app/api.py]: got it.

File: app/views.py (hunk 3 above)
Comment (alidurrani1) [reply app/views.py:137]: sure, got it.

File: app/api.py [hunk 4]
```diff
@@ -17,11 +18,34 @@ def fetch_from_api():
     """)
     url = 'https://parseapi.back4app.com/classes/Carmodels_Car_Model_List?limit=10&keys=Make,Year&where=%s' % where
     headers = {
```
Comment (maimoonaiqbal94) [inline app/api.py:20]: Can you try using python f-strings? they are more structured and easy to use.

File: app/consts.py [hunk 5]
```diff
@@ -0,0 +1,2 @@
+sender = 'alidurrani550@gmail.com'
```
Comment (maimoonaiqbal94) [inline app/consts.py:1]: pep8 issue. Please fix all of these issues from your code.

Comment (alidurrani1) [reply app/consts.py:1]: sure, boss.

File: app/api.py (hunk 4 above)
Comment (alidurrani1) [reply app/api.py:20]: sure, boss.

File: bin/celery-scheduler.sh [hunk 6]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+cd /home/ali/Desktop/Assesment/app/
```
Comment (maimoonaiqbal94) [inline bin/celery-scheduler.sh:2]: please study about these paths.. what if the code is deployed on some other machine, then this path wont work?

**Review by maimoonaiqbal94 — APPROVED** [review]

**Pull request merged**